src/modules/navigator/loiter.cpp

```diff
@@ -94,7 +94,10 @@ Loiter::set_loiter_position()
 		_mission_item.nav_cmd = NAV_CMD_IDLE;
 
 	} else {
-		if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
+		if (pos_sp_triplet->current.valid && pos_sp_triplet->current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
```

Comment:
@KonradRudin Just an idea, should we add a protection for loiter points which are too far away from the vehicle?

Reply:
hmm, i thought of making a check to only take it if we are on the loiter. So distance from vehicle to center point should be less than loiter_radius + acceptance_radius. That should do the trick, right?

Reply:
@KonradRudin Yes, that's exactly what I meant

Reply:
I have a new proposal just commited. @RomanBapst can you review again please?

Reply:
@RomanBapst ping